rackcity/models/it_instance.py

```diff
@@ -22,3 +35,11 @@ class ITInstance(models.Model):
     class Meta:
         ordering = ['hostname']
         verbose_name = 'instance'
+
+    def save(self, *args, **kwargs):
+        try:
+            validate_hostname(self.hostname)
+        except ValidationError as valid_error:
+            raise valid_error
+        else:
+            super(ITInstance, self).save(*args, **kwargs)
```

Comment:
validators are NOT called on save automatically, unfortunately

Reply:
Could we override is_valid to call this instead of save? For bulk-upload this would make the difference between failing early and rejecting the whole file, vs starting to save and then failing halfway through. The req's mention something like "reject the file if there are naming issues or other conflicts" so I think we have to fail early there.

Alternatively we could call `validate_hostname` manually in bulk.

IF WE CHANGE THIS would need to make sure it works for instances/modify because iirc we don't call is_valid there, we just rely on save for the validation